Gems/Atom/RPI/Code/Source/RPI.Builders/Model/MorphTargetExporter.cpp

```diff
@@ -159,9 +159,13 @@ namespace AZ::RPI
 
         // Determine the vertex index range for the morph target.
         const uint32_t numVertices = blendShapeData->GetVertexCount();
-        AZ_Assert(blendShapeData->GetVertexCount() == sourceMesh.m_meshData->GetVertexCount(),
-            "Blend shape (%s) contains more/less vertices (%d) than the neutral mesh (%d).",
-            blendShapeName.c_str(), numVertices, sourceMesh.m_meshData->GetVertexCount());
+        if (blendShapeData->GetVertexCount() != sourceMesh.m_meshData->GetVertexCount())
+        {
+            AZ_Error(ModelAssetBuilderComponent::s_builderName, false,
+                "Skipping blend shape (%s) as it contains more/less vertices (%d) than the neutral mesh (%d).",
```

Comment:
Maybe this message could contain information about the most common cause of this - namely a blend shape influencing multiple meshes?

Reply:
Good call. I have been thinking about this, too, but didn't go ahead to not make the error message too long as it might show up multiple times in the AP but as you're calling it out now as well, I went ahead and added it.

Reply:
Yeah, the long message is needed right now, until o3de supports multi-mesh blend-shapes. Otherwise, users might get confused why their blend shapes have incorrect counts, while in DCC utils everything 'works' :smile: